### DISCUSSION TITLE
How do I parse the HTML from the editorState without an extra editor

### DISCUSSION THREAD

**Discussion by reymon359** [facebook/lexical#3795] [Q&A]
Hey there 👋🏽 

I have a posts system and when someone submits a post I am saving the `editorState` as a JSON string in a Postgres database. 

Then, when I need to show the HTML, I am using a custom hook which loads an auxiliary editor just to obtain the HTML. 

Is this the proper way? Looks to me a bit overengineering 🤔  
I don't want to load other editor, and render it too just to obtain the HTML.

Any idea of how to improve this or suggestion for a different approach? 😄

The render post component:

```tsx
export const PostDetails = () => {
  const postSlug = useParam("postSlug", "string")
  const [postDetails] = useQuery(getPostPageDetails, { slug: postSlug })

  const { html, AuxEditor } = useGetHtmlFromState(postDetails.content as unknown as EditorState)

  return (
    <>
      <AuxEditor />
      <Text dangerouslySetInnerHTML={{ __html: html }} />
    </>
  )
}
```

The hook to get the HTML useGetHtmlFromState (it uses the same config as the input)

```tsx
export const useGetHtmlFromState = (state: EditorState) => {
  const [html, setHtml] = useState("")

  function MyCustomStateToHtmlPlugin() {
    const [editor] = useLexicalComposerContext()
    editor.update(() => {
      const html = $generateHtmlFromNodes(editor, null)
      setHtml(html)
    })

    return null
  }

  const AuxEditor = () => {
    return (
      <LexicalComposer
        initialConfig={{
          namespace: "MyEditor",
          onError: console.error,
          editorState: state,
          theme: exampleTheme,
          nodes: [
            HeadingNode,
            ListNode,
            ListItemNode,
            QuoteNode,
            CodeNode,
            CodeHighlightNode,
            TableNode,
            TableCellNode,
            TableRowNode,
            AutoLinkNode,
            LinkNode,
          ],
        }}
      >
        <MyCustomStateToHtmlPlugin />
      </LexicalComposer>
    )
  }

  return { html, AuxEditor }
}
```

**Comment by PhongLi** [comment]
Have you found any solution to this problem yet?

**Reply by reymon359** [reply]
Not a proper solution yet, sorry

**Reply by PhongLi** [reply]
So, the way you use it is still the same, right? Do you experience any performance impact or encounter any issues?

**Reply by reymon359** [reply]
Hey @PhongLi, what I am using to display the contents of the post in HTML is using the lexical editor with `editable=false`. Here is more info about it https://lexical.dev/docs/concepts/read-only#setting-the-mode. Hope it helps 😄
